subsys/net/lib/dns/resolve.c

```diff
@@ -1414,7 +1417,8 @@ int dns_resolve_name(struct dns_resolve_context *ctx,
 
 try_resolve:
 #ifdef CONFIG_DNS_RESOLVER_CACHE
-	ret = dns_cache_find(&dns_cache, query, cached_info, ARRAY_SIZE(cached_info));
+	ret = dns_cache_find(&dns_cache, query, type == DNS_QUERY_TYPE_A ? AF_INET : AF_INET6,
```

Comment:
Regarding my earlier ANY resource type, instead of the family (which is not really a DNS construct), we could give the resource type parameter to `dns_cache_find()` and then figure out the family there. Then it is easier to add the ANY support later.